### PR TITLE
Dataset metadata

### LINKED ISSUE #8687: Create dataset metadata dialog
Dialog to edit the dataset metadata like, attribution, description, license,...

### PR DESCRIPTION
**Pull request by nobuti**
This PR fixes #8687 and #8494.

metadata-dataset

cc @xavijam

### CODE REVIEW

**Comment by xavijam** [comment]
Could we center a bit more the form?

File: lib/assets/javascripts/cartodb3/components/modals/dataset-metadata/dataset-metadata-model.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      license: this._visDefinitionModel.get('license'),
+      source: this._visDefinitionModel.get('source'),
+      attributions: this._visDefinitionModel.get('attributions'),
+      tags: this._visDefinitionModel.get('tags')
+    };
+
+    this.set(data);
+
+    // Validation control variable
+    this._errors = [];
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/modals/dataset-metadata/dataset-metadata-model.js]: Backbone has `validationError` (http://backbonejs.org/#Model-validationError) variable that stores the problem for you.

File: lib/assets/javascripts/cartodb3/components/modals/dataset-metadata/dataset-metadata-view.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  render: function () {
+    this.clearSubViews();
+
+    if (this._hasError()) {
+      this._renderError();
+    } else if (this._isLoading()) {
+      this._renderLoading();
+    } else {
+      this._initViews();
+    }
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/modals/dataset-metadata/dataset-metadata-view.js]: I miss: 
```js
…
return this;
…
```

File: lib/assets/javascripts/cartodb3/components/modals/dataset-metadata/dataset-metadata-view.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  },
+
+  _initViews: function () {
+    this.$el.html(template());
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/modals/dataset-metadata/dataset-metadata-view.js]: I'll do this within the render method, not inside initViews, `manias`.

File: lib/assets/javascripts/cartodb3/components/modals/dataset-metadata/form/form-view.js [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    var title = this.$('.js-markdown').data('tooltip');
+    this._removeTooltip();
+
+    this._tooltip = new TipsyTooltipView({
+      el: this.$('.js-markdown'),
+      html: true,
+      title: function () {
+        return title;
+      }
+    });
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/modals/dataset-metadata/form/form-view.js]: You can add it to the rest of the view `this.addView(…)` and then in the render, ```this.clearSubViews();```.

File: app/assets/stylesheets/editor-3/metadata-form.css.scss [hunk 5]
```diff
@@ -20,11 +20,25 @@
...
+    max-width: 100%;
+  }
+}
+
 // To be fixed in cartoassets
 .u-grow {
   @include flex(1);
 }
 
+.Metadata-footer {
```
Comment (piensaenpixel) [inline app/assets/stylesheets/editor-3/metadata-form.css.scss]: .u-justifyEnd

**Comment by xavijam** [comment]
Several things:

- If user is not owner of the data, he couldn't open the metadata dialog.
- If dataset is synced, he couldn't change the title of the dataset.

All of that under control?

**Comment by nobuti** [comment]
Yes, the first scenario in https://github.com/CartoDB/cartodb/pull/8726/commits/1c36f5a78812572b50552404b752c196f1f27d9b and the second one in https://github.com/CartoDB/cartodb/pull/8726/commits/af7f04e504e451d42ed112d5e5a3d4da9c559391

**Comment by xavijam** [comment]
Perfect 💋

**Pull request merged**